cohttp-eio/src/server.ml

```diff
@@ -0,0 +1,260 @@
+        Scanf.bscanf in_channel "%d" (fun n ->
+            close ();
+            n)
+      with e ->
+        close ();
+        raise e)
+  | (exception Not_found)
+  | (exception Sys_error _)
+  | (exception Failure _)
+  | (exception Scanf.Scan_failure _)
+  | (exception End_of_file)
+  | (exception Unix.Unix_error (_, _, _)) ->
+      1
+
+(** [to_rfc1123 t] converts [t] to a string in a format as defined by RFC 1123. *)
+let datetime_to_string (tm : Unix.tm) =
+  let weekday =
+    match tm.tm_wday with
+    | 0 -> "Sun"
+    | 1 -> "Mon"
+    | 2 -> "Tue"
+    | 3 -> "Wed"
+    | 4 -> "Thu"
+    | 5 -> "Fri"
+    | 6 -> "Sat"
+    | 7 -> "Sun"
+    | _ -> assert false
+  in
+  let month =
+    match tm.tm_mon with
+    | 0 -> "Jan"
+    | 1 -> "Feb"
+    | 2 -> "Mar"
+    | 3 -> "Apr"
+    | 4 -> "May"
+    | 5 -> "Jun"
+    | 6 -> "Jul"
+    | 7 -> "Aug"
+    | 8 -> "Sep"
+    | 9 -> "Oct"
+    | 10 -> "Nov"
+    | 11 -> "Dec"
+    | _ -> assert false
+  in
+  Format.sprintf "%s, %02d %s %04d %02d:%02d:%02d GMT" weekday tm.tm_mday month
+    (1900 + tm.tm_year) tm.tm_hour tm.tm_min tm.tm_sec
+
+(* https://datatracker.ietf.org/doc/html/rfc7230#section-4.1 *)
+let write_chunked faraday chunk_writer =
+  let write_extensions exts =
+    List.iter
+      (fun { Chunk.name; value } ->
+        Faraday.write_char faraday ';';
+        Faraday.write_string faraday name;
+        Option.iter
+          (fun v ->
+            let v = Printf.sprintf "=%s" v in
+            Faraday.write_string faraday v)
+          value)
+      exts
+  in
+  let write = function
+    | Chunk.Chunk { size; data; extensions } ->
+        let size = Printf.sprintf "%X" size in
+        Faraday.write_string faraday size;
+        write_extensions extensions;
+        Faraday.write_string faraday "\r\n";
+        Faraday.write_bigstring faraday data.buffer;
+        Faraday.write_string faraday "\r\n"
+    | Chunk.Last_chunk extensions ->
+        let size = Printf.sprintf "%X" 0 in
+        Faraday.write_string faraday size;
+        write_extensions extensions;
+        Faraday.write_string faraday "\r\n"
+  in
+  chunk_writer write
+
+let write_response (client_conn : Client_connection.t) { Response.res; body } =
+  let faraday = Faraday.create Reader.default_io_buffer_size in
+  let serialize () =
+    let status_line =
+      let version = Http.Response.version res |> Http.Version.to_string in
+      let status = Http.Response.status res |> Http.Status.to_int in
+      let status_phrase = Http.Status.reason_phrase_of_code status in
+      Printf.sprintf "%s %d %s\r\n" version status status_phrase
+    in
+    Faraday.write_string faraday status_line;
+    let headers = Http.Response.headers res in
+    let headers =
+      (*--- Don't cache set-cookie headers in browsers and proxies. ---*)
+      if Http.Header.mem headers "set-cookie" then
+        Http.Header.add headers "Cache-Control" {|no-cache="Set-Cookie"|}
+      else headers
```

Comment:
We can just let downstream frameworks take care of this for us.

Reply:
Removed now.